src/components/Settings/Settings.js

```diff
@@ -26,22 +40,119 @@ export default class Settings extends React.Component {
             value={this.props.currentNetwork}
             onChange={this.handleChangeSelectedNetwork}
           >
-            <option value="MainNet">MainNet</option>
-            <option value="TestNet">TestNet</option>
-            <option value="CozNet">CozNet</option>
-            <option value="nOSLocal">nOS Local</option>
+            {map(settings.networks, this.renderNetworkOption)}
           </select>
         </label>
+        <div className={styles.networkDetails}>
+          Custom Neoscan URL: {this.getCurrentNetworkUrl()}
+        </div>
+        {this.renderButtons()}
         {this.state.saved && <Saved /> }
       </div>
     );
   }
 
+  renderNetworkOption = (network, key) => {
+    return (
+      <option key={key} value={network.name}>
+        {network.name}
+      </option>
+    );
+  }
+
+  renderButtons = () => {
+    return (
+      <div>
+        <Button onClick={this.handleAddNewNetwork}>
+          Add custom network configuration
+        </Button>
+        <div className={styles.divider} />
+        <Button onClick={this.handleClearNetworks}>
+          Clear custom network configurations
+        </Button>
+      </div>
+    );
+  }
+
+  handleClearNetworks = () => {
+    // If user is on a custom network, switch to TestNet
+    if (!PREDEFINED_NETWORKS.includes(this.props.currentNetwork)) {
+      this.props.setCurrentNetwork('TestNet');
```

Comment:
I've also changed this to `setCurrentNetwork` instead of `saveNetwork` because we are not saving and the `saveNetwork` code will trigger setState while the alert for clearing network is shown which causes an error `Can't call setState (or forceUpdate) on an unmounted component`